classes/task/add_meeting_registrant.php

```diff
@@ -38,62 +38,76 @@ public function execute()
         require_once($CFG->dirroot . '/mod/zoom/helperfunctions.php');
 
         $service = new \mod_zoom_webservice();
-        $queryToGetMeetingAndStudentDetails = "SELECT u.id, c.id AS 'program_id', mz.meeting_id, u.firstname, u.lastname, u.email
-        FROM mdl_user u
-                 JOIN mdl_user_enrolments ue ON ue.userid = u.id
-                 JOIN mdl_enrol e ON e.id = ue.enrolid
-                 JOIN mdl_role_assignments ra ON ra.userid = u.id
-                 JOIN mdl_context ct ON ct.id = ra.contextid AND ct.contextlevel = 50
-                 JOIN mdl_course c ON c.id = ct.instanceid and e.courseid = c.id
-                 JOIN mdl_role r ON r.id = ra.roleid AND r.shortname = 'student'
-                 JOIN mdl_zoom mz ON mz.course = c.id
-        WHERE e.status = 0 AND u.suspended = 0 AND u.deleted = 0
-        AND (ue.timeend = 0 OR ue.timeend > UNIX_TIMESTAMP(NOW())) AND ue.status = 0
-        AND mz.enable_registration = 1
-        AND NOT EXISTS(SELECT 1 FROM
-            mdl_zoom_meeting_registrant zmr
-            WHERE zmr.meeting_id = mz.meeting_id AND zmr.email = u.email)
-            LIMIT 30";
-        $meetingEnrolledUser = $DB->get_records_sql($queryToGetMeetingAndStudentDetails);
 
-        foreach ($meetingEnrolledUser as $data) {
-            try {
+        $queryToFindZoomMeetings = "SELECT mdl_zoom.meeting_id, mdl_zoom.webinar FROM mdl_zoom WHERE mdl_zoom.meeting_id NOT IN (SELECT meeting_id FROM mdl_zoom_meeting_registrant)";
+        $meetingAndWebinarList = $DB->get_records_sql($queryToFindZoomMeetings);
 
-                $response = $service->add_meeting_registrants($data->meeting_id, $data->firstname, $data->lastname, $data->email);
-                if (!empty($response)) {
+        if (!empty($meetingAndWebinarList)) {
+            foreach ($meetingAndWebinarList as $record) {
+                try {
+                    $findMeeting = $service->get_meeting_webinar_info($record->meeting_id, $record->webinar);
+                    if (!empty($findMeeting)) {
+                        $queryToGetMeetingAndStudentDetails = "SELECT u.id, c.id AS 'program_id', mz.meeting_id, u.firstname, u.lastname, u.email
+                            FROM mdl_user u
+                                    JOIN mdl_user_enrolments ue ON ue.userid = u.id
+                                    JOIN mdl_enrol e ON e.id = ue.enrolid
+                                    JOIN mdl_role_assignments ra ON ra.userid = u.id
+                                    JOIN mdl_context ct ON ct.id = ra.contextid AND ct.contextlevel = 50
+                                    JOIN mdl_course c ON c.id = ct.instanceid and e.courseid = c.id
+                                    JOIN mdl_role r ON r.id = ra.roleid AND r.shortname = 'student'
+                                    JOIN mdl_zoom mz ON mz.course = c.id
+                            WHERE e.status = 0 AND u.suspended = 0 AND u.deleted = 0
```

Comment:
@Ash-5 I know that this was already existing query. Just wanted to know of it is better to move the where conditions to the corresponding joins instead of here?

Reply:
@meyyappanv I have done the changes can you please check again.